Backend/spotifyRoutes.js

```diff
@@ -358,4 +360,74 @@ router.get("/compatibility", async (req, res) => {
   }
 });
 
+function calcJaccard(intersection, union) {
+    if (intersection.size === 0) {
+        return 0;
+    } else {
+        let intersectionNum = 0, unionNum = 0;
+        for (const song of intersection) {
+            const [title, popularity] = song.split(":");
+            if (0 <= popularity && popularity <= 34 ) {
+                intersectionNum += 3
+            }else if (35 <= popularity && popularity <= 68) {
+                intersectionNum += 2
+            }else if (69 <= popularity && popularity <= 100) {
+                intersectionNum += 1
+            }
+        }
+
+        for (const song of union) {
+            const [title, popularity] = song.split(":");
+            if (0 <= popularity && popularity <= 34 ) {
+                unionNum += 3
+            }else if (35 <= popularity && popularity <= 68) {
+                unionNum += 2
+            }else if (69 <= popularity && popularity <= 100) {
+                unionNum += 1
+            }
+        }
+        const index = (intersectionNum / unionNum).toFixed(3)
+        return parseInt(index)
+    }
+}
+
+async function calcTopSongsJaccard(currentUserData, currentFriendData) {
```

Comment:
let's create one function for all these calculations, and pass in the sets we want to find the union between